### PR TITLE
Update readme for DPS support

### PR DESCRIPTION
**Pull request by yzhong94**
Add feature matrix for DPS device SDK

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -51,6 +51,17 @@ git clone --recursive https://github.com/Azure/azure-iot-sdk-csharp.git
...
+| TPM Individual Enrollment   | :heavy_check_mark: | :heavy_check_mark: | :heavy_multiplication_x: | :heavy_check_mark: | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [Trusted Platform Module](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#trusted-platform-module-tpm).  Please review the [samples](./provisioning/device/samples/) folder on how to create a device client.    Websocket connection over AMQP is in progress.                                                                                                                                                                                                        |
```
Comment (tameraw) [inline readme.md]: >:heavy_check_mark: [](start = 101, length = 18)

This is also not working.

File: readme.md [hunk 2]
```diff
@@ -51,6 +51,17 @@ git clone --recursive https://github.com/Azure/azure-iot-sdk-csharp.git
...
+| X.509 Individual Enrollment | :heavy_check_mark: | :heavy_check_mark:* | :heavy_check_mark: | :heavy_check_mark:* | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [X.509 root certificate](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#root-certificate).  Please review the [samples](./provisioning/device/samples/) folder on how to create a device client.    Websocket connection with X.509 on .NETCore is available on Linux but not on Windows.  For Windows, please use .Net instead.  |
```
Comment (tameraw) [inline readme.md]: >.Net [](start = 650, length = 4)

.NETFramework

Comment (tameraw) [inline readme.md]: >instead [](start = 655, length = 7)

Also, Chris will provide you a link to the blocking issue in .NETCore.

File: readme.md [hunk 3]
```diff
@@ -51,6 +51,17 @@ git clone --recursive https://github.com/Azure/azure-iot-sdk-csharp.git
...
+| X.509 Enrollment Group      | :heavy_check_mark: | :heavy_check_mark:* | :heavy_check_mark: | :heavy_check_mark:* | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [X.509 leaf certificate](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#leaf-certificate)).  Please review the [samples](./provisioning/device/samples/) folder on how to create a device client.    Websocket connection with X.509 on .NETCore is available on Linux but not on Windows.  For Windows, please use .Net instead.                                                                                                                                                                                          |
```
Comment (CIPop) [inline readme.md]: Please reference our dependency here: https://github.com/dotnet/corefx/wiki/ApiCompat#clientwebsocketoptions

Comment (tameraw) [inline readme.md]: >.Net instead.   [](start = 651, length = 15)

Same here.

File: readme.md (hunk 2 above)
Comment (CIPop) [reply readme.md]: Please reference our dependency here: https://github.com/dotnet/corefx/wiki/ApiCompat#clientwebsocketoptions
The bug is within the wiki.

File: readme.md (hunk 1 above)
Comment (CIPop) [reply readme.md]: @yzhong94 @tameraw TPM over AMQP works (just tested) - sorry for giving you the wrong info earlier today.
/cc @jasmineymlo

**Comment by yzhong94** [comment]
@CIPop Please merge this!

**Pull request merged**